test/distributed/_sharded_tensor/test_sharded_tensor.py

```diff
@@ -329,6 +330,187 @@ def test_shard_parameter_errors(self):
             shard_parameter(fc, 'weight', spec)
 
 
+class TestCollectLocalShard(ShardedTensorTestBase):
+    @with_comms(init_rpc=False)
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_collect_local_shard(self):
+        spec = ChunkShardingSpec(
+            dim=0,
+            placements=[
+                "rank:0/cuda:0",
+                "rank:1/cuda:1",
+                "rank:2/cuda:2",
+                "rank:3/cuda:3",
+            ],
+        )
+        st = _sharded_tensor.rand(spec, 24, 12)
+        local_shard = st.collect_local_shard()
+        self.assertEqual(torch.Size([6, 12]), local_shard.size())
+        self.assertEqual(
+            st.local_shards()[0].tensor,
+            local_shard,
+        )
+
+    @with_comms(init_rpc=False)
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_collect_local_shard_error(self):
+        spec = ChunkShardingSpec(
+            dim=0,
+            placements=[
+                "rank:0/cuda:0",
+                "rank:0/cuda:0",
+                "rank:1/cuda:1",
+                "rank:1/cuda:1",
+                "rank:1/cuda:1",
+                "rank:2/cuda:2",
+                "rank:2/cuda:2",
+                "rank:2/cuda:2",
+                "rank:3/cuda:3",
+                "rank:3/cuda:3",
+            ],
+        )
+        st = _sharded_tensor.rand(spec, 24, 12)
+        with self.assertRaisesRegex(
+            NotImplementedError, "Only single local shard is supported."
+        ):
+            local_shard = st.collect_local_shard()
+
+
+class TestReshard(ShardedTensorTestBase):
+    def _run_sharded_tensor_reshard(self, sharding_spec, reshard_spec, input_size):
+        torch.manual_seed(0)
+        fc = torch.nn.Linear(*input_size).cuda(self.rank)
+        fc_compare = torch.nn.Linear(*input_size).cuda(self.rank)
+        fc_compare.weight = torch.nn.Parameter(fc.weight.detach().clone())
+        shard_parameter(fc, "weight", sharding_spec)
+        st = fc.weight
+        st.reshard(reshard_spec)
+        shard_parameter(fc_compare, "weight", reshard_spec)
+        st_compare = fc_compare.weight
```

Comment:
Do we need all this if we are just testing ShardedTensor reshard? We can just create a ShardedTensor directly instead of using nn.Linear and `shard_parameter`?

Reply:
Sure. I was taking a short-cut here. Let me change it.

Reply:
Update: Will create a new API for tensor sharding.